tests/unittest/disaggregated/test_mamba_transfer.py

```
@@ -348,6 +354,132 @@ def test_mamba_receiver_payload_bytes_matched_tp():
+    assert mapped.src.memory.ptrs.tolist() == [1000 + 512 + 32]
+    assert mapped.dst.memory.ptrs.tolist() == [10000 + 512 + 2 * 32]
+    assert mapped.src.memory.bytes_per_region == 8
+    assert peer.mamba_receiver_payload_bytes(self_page_table, peer_page_table, dst_slot=2) == 72
+
+
+def test_mamba_policy_rejects_incompatible_recurrent_side_state():
+    self_page_table = _make_mamba_page_table(base_address=1000)
+    peer_page_table = _make_mamba_page_table(base_address=10000, side_bytes=16)
+    rank_info = _make_rank_info()
+
+    with pytest.raises(ValueError, match="side-state role.*slot_bytes differs"):
```

Ruff (0.16.2)

[warning] Pattern passed to `match=` contains metacharacters but is neither escaped nor raw

(RUF043)